projects/com.oracle.truffle.llvm.parser.impl/src/com/oracle/truffle/llvm/parser/impl/LLVMVisitor.java

```diff
@@ -1048,6 +1048,18 @@ private LLVMExpressionNode visitConstant(EObject type, Constant constant) throws
                 GlobalVariable globalVariable = (GlobalVariable) constant.getRef();
                 String globalVarName = globalVariable.getName();
                 String linkage = globalVariable.getLinkage();
+
+                /*
+                 * Global variables in C can be 'extern', but still managed because they're defined
+                 * in another managed file. We should be able to detect that by looking for existing
+                 * global variables with the same name, so I would think that we should just need to
+                 * add
+                 * 
+                 * && !globalVars.containsKey(globalVariable)
+                 * 
+                 * to the condition below, but if I do that things go pretty badly wrong and we end
+                 * up crashing.
```

Comment:
Is the original code wrong anyway then? In your example it will try to look up `a` using NFI, but if the variable is defined in LLVM IR (in the same file or not) it won't be found like that will it?